### ISSUE TITLE
Organization Bypass on Login via WebUI

### ISSUE THREAD

**Issue by mhwest13** [n4bb12/verdaccio-github-oauth-ui#12]
## Bug Report
Forgive me if this isn't the right place to ask, I'm just trying to better understand this plugin and how it interacts with the verdaccio UI and if I'm possibly misunderstanding how this works.

#### Versions

|                           | Version |
| ------------------------- | ------- |
| verdaccio                 | 3.8.3        |
| verdaccio-github-oauth-ui | 1.5.0        |

#### Expected behavior
So I have the plugin working, enough where I can login as a user via a web browser to verdaccio, which is being told to only allow users from a specified Github Organization. I'm able to successfully whoami from the cli as well with that user. So kudos for making this actually fairly simple and straight forward.

However, when I switch to a user not in the specified Github Organization, I'm still able to authenticate into the UI via a web browser and see packages inside of the verdaccio repo. I'm not able to successfully whoami via the cli with this user, which is to be expected, although I don't know if the error being returned is exactly the one I expect.

What I do expect is for the first user which is in part of the Organization to be able to access the UI and authenticate via CLI; just as it is. However, I don't expect the second user to be able to authenticate into the UI and see packages inside of the verdaccio repo, in either a browser or via cli.

#### Observed behaviour
A user not in the specified Github Organization is able to successfully authenticate in and see packages listed inside of the verdaccio repo in a browser.

#### Steps to reproduce
1. Setup verdaccio-github-oauth-ui to restrict auth access to an org
2. Attempt to authenticate with a user not in that specified org via web browser
3. You should be able to successfully authenticate and see packages listed inside of the repo (at least I was)

#### Additional context
I have tried setting access to $authenticated in this setup as well, instead of $all, but that hasn't seem to impact my ability to login as a user not in the specified org.

This is my config file.
```
#
storage: /opt/verdaccio/cache

web:
  title: Verdaccio NPM Registry

auth:
  github-oauth-ui:
    org: Someorganization

uplinks:
  npmjs:
    url: https://registry.npmjs.org/

packages:
  '@*/*':
    # scoped packages
    access: $authenticated
    publish: $authenticated
    proxy: npmjs

  '**':
    access: $authenticated
    publish: $authenticated
    proxy: npmjs

middlewares:
  audit:
    enabled: true
  github-oauth-ui:
    client-id: github-client-id
    client-secret: github-client-secret

logs:
  - {type: file, path: /var/log/verdaccio/verdaccio.log, level: info}

listen: 0.0.0.0:4873

```
This is the good cli output from my user which should work.

```
npm info it worked if it ends with ok
npm verb cli [ '/Users/joebob/.nvm/versions/node/v8.11.3/bin/node',
npm verb cli   '/Users/joebob/.nvm/versions/node/v8.11.3/bin/npm',
npm verb cli   '-dd',
npm verb cli   'whoami',
npm verb cli   '--registry',
npm verb cli   'https://verdaccio.some.registry' ]
npm info using npm@6.1.0
npm info using node@v8.11.3
npm verb request uri https://verdaccio.some.registry/-/whoami
npm verb request always-auth set; sending authorization
npm info attempt registry request try #1 at 12:20:24
npm verb request using bearer token for auth
npm verb request id 3eedb60a6e4a2691
npm http request GET https://verdaccio.some.registry/-/whoami
npm http 200 https://verdaccio.some.registry/-/whoami
npm verb headers { 'access-control-allow-origin': '*',
npm verb headers   'content-type': 'application/json; charset=utf-8',
npm verb headers   date: 'Thu, 11 Oct 2018 19:20:24 GMT',
npm verb headers   etag: '"f4ff006b5175c72c079cb9583afc1394"',
npm verb headers   vary: 'Accept-Encoding',
npm verb headers   'x-powered-by': 'verdaccio/3.8.3',
npm verb headers   'x-status-cat': 'http://flic.kr/p/aVuVsF',
npm verb headers   'content-length': '29',
npm verb headers   connection: 'keep-alive' }
mhwest13
npm verb exit [ 0, true ]
npm timing npm Completed in 608ms
npm info ok
```

This is the bad cli output from the user that shouldn't work; I guess I actually expected more like a 403 to be returned, not an invalid auth token message.

```
npm info it worked if it ends with ok
npm verb cli [ '/Users/joebob/.nvm/versions/node/v8.11.3/bin/node',
npm verb cli   '/Users/joebob/.nvm/versions/node/v8.11.3/bin/npm',
npm verb cli   '-dd',
npm verb cli   'whoami',
npm verb cli   '--registry',
npm verb cli   'https://verdaccio.some.registry' ]
npm info using npm@6.1.0
npm info using node@v8.11.3
npm verb request uri https://verdaccio.some.registry/-/whoami
npm verb request always-auth set; sending authorization
npm info attempt registry request try #1 at 12:16:03
npm verb request using bearer token for auth
npm verb request id 93b3eb6268b44496
npm http request GET https://verdaccio.some.registry/-/whoami
npm http 200 https://verdaccio.some.registry/-/whoami
npm verb headers { 'access-control-allow-origin': '*',
npm verb headers   'content-type': 'application/json; charset=utf-8',
npm verb headers   date: 'Thu, 11 Oct 2018 19:16:03 GMT',
npm verb headers   etag: '"8a80554c91d9fca8acb82f023de02f11"',
npm verb headers   vary: 'Accept-Encoding',
npm verb headers   'x-powered-by': 'verdaccio/3.8.3',
npm verb headers   'x-status-cat': 'http://flic.kr/p/aVuVsF',
npm verb headers   'content-length': '3',
npm verb headers   connection: 'keep-alive' }
npm verb stack Error: Your auth token is no longer valid. Please log in again.
npm verb stack     at /Users/joebob/.nvm/versions/node/v8.11.3/lib/node_modules/npm/lib/whoami.js:27:32
npm verb stack     at /Users/joebob/.nvm/versions/node/v8.11.3/lib/node_modules/npm/node_modules/npm-registry-client/lib/whoami.js:19:5
npm verb stack     at f (/Users/joebob/.nvm/versions/node/v8.11.3/lib/node_modules/npm/node_modules/once/once.js:25:25)
npm verb stack     at /Users/joebob/.nvm/versions/node/v8.11.3/lib/node_modules/npm/node_modules/npm-registry-client/lib/request.js:78:16
npm verb stack     at /Users/joebob/.nvm/versions/node/v8.11.3/lib/node_modules/npm/node_modules/npm-registry-client/lib/request.js:105:12
npm verb stack     at f (/Users/joebob/.nvm/versions/node/v8.11.3/lib/node_modules/npm/node_modules/once/once.js:25:25)
npm verb stack     at RegClient.<anonymous> (/Users/joebob/.nvm/versions/node/v8.11.3/lib/node_modules/npm/node_modules/npm-registry-client/lib/request.js:311:12)
npm verb stack     at Request._callback (/Users/joebob/.nvm/versions/node/v8.11.3/lib/node_modules/npm/node_modules/npm-registry-client/lib/request.js:216:14)
npm verb stack     at Request.self.callback (/Users/joebob/.nvm/versions/node/v8.11.3/lib/node_modules/npm/node_modules/request/request.js:185:22)
npm verb stack     at emitTwo (events.js:126:13)
npm verb cwd /Users/joebob/code/bots/slackbot
npm verb Darwin 16.7.0
npm verb argv "/Users/joebob/.nvm/versions/node/v8.11.3/bin/node" "/Users/joebob/.nvm/versions/node/v8.11.3/bin/npm" "-dd" "whoami" "--registry" "https://verdaccio.some.registry"
npm verb node v8.11.3
npm verb npm  v6.1.0
npm ERR! code ENEEDAUTH
npm ERR! need auth Your auth token is no longer valid. Please log in again.
npm ERR! need auth You need to authorize this machine using `npm adduser`
npm verb exit [ 1, true ]
npm timing npm Completed in 614ms
```

**Comment by mhwest13** [comment]
I figured I'd highlight that the organization is the one that own's this OAuth application inside of Github, and we have restricted access to third parties to this Github Organization. From my understanding, since the organization owns the application, it is already given access to read organizational information.

Also, during the initial login flow; the Github OAuth landing page, does indicate that the application is requiring the ability to read my teams and organizations.
